src/overlays/actors/ovl_En_Tk/z_en_tk.c

```diff
@@ -35,130 +93,1379 @@ const ActorInit En_Tk_InitVars = {
+        func_800BBFB0(globalCtx, &this->unk_2B4, 10.0f, 2, 50, 30, 1);
+    }
+
+    if (func_801378B8(&this->skelAnime, 33.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_DIG_UP);
+    }
+
+    if (!(this->unk_2CA & 0x20)) {
+        if (func_801378B8(&this->skelAnime, 37.0f)) {
+            phi_s0 = NULL;
+            do {
+                phi_s0 = func_ActorCategoryIterateById(globalCtx, phi_s0, ACTORCAT_PROP, ACTOR_EN_BIGPO);
+                
+                if (phi_s0 != NULL) {
+                    if ((phi_s0->params == 3) && (Actor_DistanceBetweenActors(&this->actor, phi_s0) < 80.0f)) {
+                        phi_s0->params = 4;
+                        this->unk_2CA |= 0x20;
+                        this->unk_2E4++;
+                    }
+                    phi_s0 = phi_s0->next;
+                }
+            } while (phi_s0 != NULL);
+        }
+    }
+
+    if (func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount)) {
+        func_80AEE650(this, globalCtx);
+    }
+}
+
+void func_80AEE650(EnTk* this, GlobalContext* globalCtx) {
+    if (this->unk_36C < ARRAY_COUNT(this->unk_324)) {
+        Math_Vec3f_Copy(&this->unk_324[this->unk_36C], &this->actor.world.pos);
+        this->unk_36C++;
+    }
+    this->actionFunc = func_80AEE6B8;
+}
+
+void func_80AEE6B8(EnTk* this, GlobalContext* globalCtx) {
+    if (this->unk_2CA & 0x20) {
+        if (this->unk_2E4 >= 3) {
+            ActorCutscene_Stop(this->cutscenes[0]);
+            func_801477B4(globalCtx);
+            func_80AEDBEC(this, globalCtx);
+        } else if (func_8013E2D4(&this->actor, 0x7C, this->cutscenes[0], 0)) {
+            this->unk_310 = 3;
+            func_80AEDE10(this, globalCtx);
+            this->unk_2CA &= ~0x20;
+        }
+    } else if (func_8013E2D4(&this->actor, 0x7C, this->cutscenes[0], 0)) {
+        this->unk_310 = 4;
+        func_80AEDE10(this, globalCtx);
+    }
+}
+
+void func_80AEE784(EnTk* this, GlobalContext* globalCtx) {
+    this->unk_2D0 = -1;
+    this->unk_310 = 1;
+    Math_Vec3s_Copy(&this->actor.world.rot, &this->actor.shape.rot);
+    func_80AEEAD4(this, globalCtx);
+    this->actionFunc = func_80AEE9B0;
+}
+
+s32 func_80AEE7E0(Vec3f* arg0, f32 arg1, Vec3f* arg2, s32 arg3) {
+    s32 i;
+    s32 ret = true;
+
+    for (i = 0; i < arg3; i++) {
+        if (Math_Vec3f_DistXZ(arg0, &arg2[i]) < arg1) {
+            ret = false;
+            break;
+        }
+    }
+    return ret;
+}
+
+s32 func_80AEE86C(EnTk* this, GlobalContext* globalCtx) {
+    static Vec3f D_80AEFA78 = { 0.0f, 20.0f, 32.0f };
+    s32 pad;
+    s32 ret = false;
+    s32 pad2;
+    CollisionPoly* sp38;
+    s32 sp34;
+    Vec3f sp28;
+
+    Lib_Vec3f_TranslateAndRotateY(&this->actor.world.pos, this->actor.shape.rot.y, &D_80AEFA78, &sp28);
+    if ((func_800C40B4(&globalCtx->colCtx, &sp38, &sp34, &sp28) != BGCHECK_Y_MIN) && 
+        (func_800C9BB8(&globalCtx->colCtx, sp38, sp34) == 1U) && 
+        (this->unk_2D0 == 1U) && 
```

Comment:
```suggestion
        (func_800C9BB8(&globalCtx->colCtx, sp38, sp34) == 1) && 
        (this->unk_2D0 == 1) && 
```
Needed?

Reply:
Yeah it's needed. Whenever I add Us they're always needed, since m2c doesn't output them as I always use --no-casts.